Makefile

```diff
@@ -62,19 +62,28 @@ GOLDFLAGS ?=
 KUBE_GOLDFLAGS = $(GOLDFLAGS)
 export KUBE_GOLDFLAGS GOLDFLAGS
 
+GOGCFLAGS ?=
+KUBE_GOGCFLAGS = $(GOGCFLAGS)
+export KUBE_GOGCFLAGS GOGCFLAGS
+
 # Build code.
 #
 # Args:
 #   WHAT: Directory names to build.  If any of these directories has a 'main'
 #     package, the build will produce executable files under $(OUT_DIR)/go/bin.
 #     If not specified, "everything" will be built.
 #   GOFLAGS: Extra flags to pass to 'go' when building.
-#   GOLDFLAGS: Extra linking flags to pass to 'go' when building.
+#   GOLDFLAGS: Extra linking flags passed to 'go' when building.
+#   GOGCFLAGS: Additional go compile flags passed to 'go' when building.
 #
 # Example:
 #   make
 #   make all
 #   make all WHAT=cmd/kubelet GOFLAGS=-v
+#   make all GOGCFLAGS="-N -l"
+#     Note: Use the -N -l options to disable compiler optimizations an inlining.
```

Comment:
Interesting question.  

Thinking out loud... Another popular GOGCFLAGS pattern will be `-m -l` for performing analysis of garbage collection escapes that map to heap allocations.  Still other programmers might want to implement custom go compiler extensions.  make ESC_ANALYSIS=1 has less curb appeal.. 

Adding a make dbg target or a KUBE_DEBUG=1 variable.. That sounds like a higher level decision.  Probably a feature.. Where the GOGCFLAGS variable was just enabling the passing of go compiler flags as already done for linker flags, arguably not a feature :-) 

I suggest proposing a feature for your make * DBG=1  idea and seeing if anyone has ideas/wants more than just the gcflags compiler optimization flags for full fledged debug/support builds (runs slower but is easier to debug/support).